File: robocode2-bot-api-java/src/main/java/net/robocode2/BulletState.java
package net.robocode2;

import lombok.Builder;
import lombok.Value;

/** Current bullet state */
@Value
@Builder
public final class BulletState {
    /** ID of the bullet */
    int bulletId;
    /** ID of the bot that fired the bullet */
    int ownerId;
    /** Bullet firepower level */
    double power;
    /** X coordinate */
    double x;
    /** Y coordinate */
    double y;
    /** Direction in degrees */
    double direction;
    /** Speed measured in pixels per turn */
    double speed;
}

File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/event/BulletState.kt
package net.robocode2.gui.model.event

import net.robocode2.gui.model.types.Point

data class BulletState(
        val bulletId: Int,
        val ownerId: Int,
        val power: Double,
        val position: Point,
        val direction: Double,
        val speed: Double
)